crates/journald/lib.rs

```diff
@@ -0,0 +1,228 @@
+//! Journald drain fro slog-rs
+//!
+//! Since Journald supports structured data, structured data passed to slog is
+//! simply forwarded to Journald as structured data.
+//!
+//! # Examples
+//! ```
+//! #[macro_use]
+//! extern crate slog;
+//! extern crate slog_journald;
+//!
+//! use slog::*;
+//! use slog_journald::*;
+//!
+//! fn main() {
+//!     let root = Logger::root(JournaldDrain.fuse(), o!("build-di" => "12344"));
+//!     info!(root, "Testing journald"; "foo" => "bar");
+//! }
+//! ```
+
+#![warn(missing_docs)]
+
+extern crate libsystemd_sys;
+extern crate libc;
+extern crate slog;
+
+use std::ascii::AsciiExt;
+use std::fmt::Display;
+use std::os::raw::{c_int, c_void};
+
+use libc::{LOG_CRIT, LOG_ERR, LOG_INFO, LOG_WARNING, LOG_NOTICE, LOG_DEBUG, size_t};
+use libsystemd_sys::const_iovec;
+use libsystemd_sys::journal::sd_journal_sendv;
+use slog::{Drain, Record, OwnedKeyValueList, Level};
+use slog::ser::Result as SerResult;
+
+/// Drain records and send to journald as structured data.
+///
+/// Journald requires keys to be uppercase alphanumeric, so logging keys
+/// are capitalized and all non-alpha-numeric letters are converted to underscores.
+pub struct JournaldDrain;
+
+impl Drain for JournaldDrain {
+    type Error = ::Error;
+    fn log(&self, info: &Record, logger_values: &OwnedKeyValueList) -> Result<(), ::Error> {
+        let mut serializer = Serializer::new();
+        serializer.add_field("PRIORITY", level_to_priority(info.level()));
+        serializer.add_field("MESSAGE", info.msg());
+        serializer.add_field("CODE_FILE", info.file());
+        serializer.add_field("CODE_LINE", info.line());
+        serializer.add_field("CODE_MODULE", info.module());
+        serializer.add_field("CODE_FUNCTION", info.function());
+        for &(ref k, ref v) in logger_values.iter() {
```

Comment:
I have broke this in master (soon to be released as slog 1.1), see CHANGELOG.md. Sorry for that - it was for a good reason, and I think this is the first 3rd-party serializing drain. Can you try against `master` branch?